### PR TITLE
ft(userPage):A buyer should be able to checkout

### PR DESCRIPTION
**Pull request by Ke-bean**
## ℹ️ Description

This feature should allow a user to proceed to the next step which is completing the payment process.

## 🧪 How can this be tested?

This can be tested via the browser where the buyer can open the cart, click on proceed to checkout button , enter order information and create an order

## 📸 Screenshots

Screen Shot 2024-07-16 at 09 59 09
Screen Shot 2024-07-16 at 09 59 51
Screen Shot 2024-07-18 at 12 06 26
Screen Shot 2024-07-18 at 12 06 48
Screen Shot 2024-07-18 at 12 07 17

## 📍 Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/service/productApi.tsx
Comment (Oliviier-dev) [inline src/service/productApi.tsx:1]: You can move this to the existing file with product endpoints located at src/service/productApi.ts

File: src/components/rootcomponents/CheckoutModal.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  return (
+    <div className="fixed inset-0 bg-gray-800 bg-opacity-50 flex justify-center items-center">
+      <div className="bg-white p-8 rounded-lg shadow-lg w-full max-w-2xl">
+        <h2 className="text-2xl font-semibold mb-6 text-center">Checkout</h2>
+        <form onSubmit={handleSubmit} className="space-y-4">
+          <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
+            <div>
+              <label className="block text-sm font-medium text-gray-700 mb-1">
+                Address
+              </label>
```
Comment (Oliviier-dev) [inline src/components/rootcomponents/CheckoutModal.tsx]: use an existing label component here

File: src/components/rootcomponents/CheckoutModal.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                placeholder="Address"
+                value={formData.address}
+                onChange={handleChange}
+                className="w-full p-2 border border-gray-300 rounded-md focus:ring-blue-500 focus:border-blue-500"
+              />
+            </div>
+            <div>
+              <label className="block text-sm font-medium text-gray-700 mb-1">
+                Country
+              </label>
```
Comment (Oliviier-dev) [inline src/components/rootcomponents/CheckoutModal.tsx]: here as well, use an existing label component

File: src/components/rootcomponents/CheckoutModal.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                placeholder="Country"
+                value={formData.country}
+                onChange={handleChange}
+                className="w-full p-2 border border-gray-300 rounded-md focus:ring-blue-500 focus:border-blue-500"
+              />
+            </div>
+            <div>
+              <label className="block text-sm font-medium text-gray-700 mb-1">
+                City
+              </label>
```
Comment (Oliviier-dev) [inline src/components/rootcomponents/CheckoutModal.tsx]: and here

**Review by Oliviier-dev — COMMENTED** [review]
Nice work so far @Ke-bean look into these minor suggestions:

**Review by Oliviier-dev — DISMISSED** [review]
LGTM

**Review by martinemahirwe — DISMISSED** [review]
LGTM

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Have you really worked on the task as the description describes!! @Ke-bean 

Screenshot 2024-07-16 at 12 32 33

**Comment by Ke-bean** [comment]
> > Have you really worked on the task as the description describes!! @Ke-bean
> > Screenshot 2024-07-16 at 12 32 33
> 
> I figured out that I miss to redirect the customer to the payment portal but for the payment, it's a different task which is under development

**Comment by Ke-bean** [comment]
> Have you really worked on the task as the description describes!! @Ke-bean
> 
> Screenshot 2024-07-16 at 12 32 33

I figured out that I missed the redirection of customer to the payment portal, but integrating payment is a different task which is under development

**Review by martinemahirwe — APPROVED** [review]
LGTM

**Review by Oliviier-dev — APPROVED** [review]
LGTM

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**